cli/src/semgrep/output.py

```diff
@@ -450,85 +511,81 @@ def _post_output(self, output_url: str, output: str) -> None:
         except requests.exceptions.Timeout:
             raise SemgrepError(f"posting output to {output_url} timed out")
 
-    def _build_output(self) -> str:
-        # CliOutputExtra members
-        cli_paths = out.ScannedAndSkipped(
-            # This is incorrect when some rules are skipped by semgrep-core
-            # e.g. proprietary rules.
-            # TODO: Use what semgrep-core returns for 'scanned' and 'skipped'.
-            scanned=[out.Fpath(str(path)) for path in sorted(self.all_targets)],
-            skipped=None,
-        )
-        cli_timing: Optional[out.Profile] = None
-
-        explanations: Optional[List[out.MatchingExplanation]] = self.explanations
-
-        # Extra, extra! This just in! 🗞️
-        # The extra dict is for blatantly skipping type checking and function signatures.
-        # - The text formatter uses it to store settings
-        # You should use CliOutputExtra for better type checking
-        extra: Dict[str, Any] = {}
-        if self.settings.output_time and self.extra and self.extra.core.time:
-            cli_timing = _build_time_json(
-                self.filtered_rules,
-                self.all_targets,
-                self.extra.core.time,
-                self.profiler,
-            )
-        if self.settings.verbose_errors:
-            # TODO: use SkippedTarget directly in ignore_log or in yield_json_objects at least
-            skipped = sorted(
-                self.ignore_log.yield_json_objects(), key=lambda x: Path(x["path"])
+    def _build_output(self) -> Iterator[Tuple[Optional[str], str]]:
```

Comment:
Optional: this is probably more about the limitations on how github diffs the changes. Once things got indented, now it looks like a big change, when I think it's just a couple of small tweaks.

I think the changes would be easier to read if you have `_build_outputs` do the for loop and yielding, and `_build_output` just return the pair (output_destination, output_format).

In general, I like it more when indented blocks aren't too long. :)

Your call.